### PR TITLE
 Use proper labels for plotting if IDs are available in ReturnData

### LINKED ISSUE #2175: Use proper labels for plotting if IDs are available in ReturnData
`amici.plotting.*` uses some default labels (e.g. x1, x2, x3) if `model=None`. However, in some cases (ReturnDataReportingMode == full), we can still access IDs via the provided `rdata` argument, similar to what is done in 
https://github.com/AMICI-dev/AMICI/blob/847a5966cad44a4c3b353d1cdca520b7e83c4da9/python/sdist/amici/numpy.py#L297
In such a case, those IDs should be used to create more informative plots.

### PR DESCRIPTION
**Pull request by dweindl**
`amici.plotting.*` uses some default labels (e.g. x1, x2, x3) if `model=None`. However, in some cases (ReturnDataReportingMode == full), we can still access IDs via the provided `rdata` argument. Try that.

Closes #2175

### CODE REVIEW

File: python/sdist/amici/plotting.py [hunk 1]
```diff
@@ -64,38 +78,54 @@ def plot_observable_trajectories(
...
+
+    if marker is None:
+        # Show marker if only one time point is available,
+        #  otherwise nothing will be shown
+        marker = "o" if len(rdata.t) == 1 else None
+
+    if model is None and rdata.ptr.observable_ids is None:
+        labels = [f"$y_{{{iy}}}$" for iy in observable_indices]
+    elif model is not None and prefer_names:
+        labels = np.asarray(model.getObservableNames())[
```
Comment (FFroehlich) [inline python/sdist/amici/plotting.py:115]: unindexed_observables = model.getObservableNames() if prefer_names else model.getObservableIds()

Comment (FFroehlich) [inline python/sdist/amici/plotting.py:115]: `unindexed_observables = model.getObservableNames() if prefer_names else model.getObservableIds()` to merge `elif model is not None and prefer_names` and `elif model is not None` cases?

File: python/sdist/amici/plotting.py [hunk 2]
```diff
@@ -64,38 +78,54 @@ def plot_observable_trajectories(
...
+            list(observable_indices)
+        ]
+        labels = [
+            l if l else model.getObservableNames()[ix]
```
Comment (FFroehlich) [inline python/sdist/amici/plotting.py:119]:
```suggestion
            l if l else model.getObservableIds()[ix]
```

File: python/sdist/amici/plotting.py [hunk 3]
```diff
@@ -16,42 +17,55 @@
...
+
+    if marker is None:
+        # Show marker if only one time point is available,
+        #  otherwise nothing will be shown
+        marker = "o" if len(rdata.t) == 1 else None
+
+    if model is None and rdata.ptr.state_ids is None:
+        labels = [f"$x_{{{ix}}}$" for ix in state_indices]
+    elif model is not None and prefer_names:
+        labels = np.asarray(model.getStateNames())[list(state_indices)]
```
Comment (FFroehlich) [inline python/sdist/amici/plotting.py:57]: see below

File: python/sdist/amici/plotting.py [hunk 4]
```diff
@@ -16,42 +17,55 @@
...
+        labels = [
+            l if l else model.getStateNames()[ix]
```
Comment (FFroehlich) [inline python/sdist/amici/plotting.py:59]:
```suggestion
            l if l else model.getStateIds()[ix]
```

**Review by FFroehlich — APPROVED** [review]
👍

File: python/sdist/amici/plotting.py (hunk 1 above)
Comment (dweindl) [reply python/sdist/amici/plotting.py:115]: I would rather keep it separate as it's already sufficiently confusing what's to be used :sweat_smile:.
We'd still need some distinction between the two cases to fall back to IDs if the name is ''.

**Pull request merged**